### DISCUSSION TITLE
Need help with .hx Video Stuff

### DISCUSSION THREAD

**Discussion by THEkatinamicrowave** [CodenameCrew/CodenameEngine#298] [Q&A] [answered]
I was making a cutscene intro for the game so that after the BetaWarning Screen, it transitions to an MP4 (which I made a separate .hx file for). Unfortunately, I'm stupid and only learned LUA so I had to go off of existing .hx files and improvise by making educated guesses of what each one line of code did. Using that, I made, with some help from Nex, this beauty of a script (/s):

```haxe
package funkin.menus;

import funkin.game.cutscenes.ScriptedCutscene;

class PeterCutsceneIntro extends ScriptedCutscene {
    
    public function new() {
        super(Paths.video("betergriffin"), function() {
            FlxG.switchState(new TitleState());
        });
    };

}
```
Surely nothing will go wrong...

![image](https://github.com/FNF-CNE-Devs/CodenameEngine/assets/89645987/76b7f3bb-a62a-4d5e-b4aa-ed03d8c9e90e)

If anybody is intelligent, please help me out here.

**Comment by r6915ee** [comment]
According to the script you provided, it seems you're trying to access a video file. In that case, it would be better to extend the `VideoCutscene` class instead. This is so there is no need to use HScript, since the first argument in the `ScriptedCutscene` class's `new()` function uses the `Script` class.

Following this, change the `super()` function in the state accordingly. You should get something like:

`super(Paths.video("betergriffin", "mp4"), FlxG.switchState(new TitleState()))`

Changing this is necessary, because the `super()` function essentially IS the `new()` function for any class that is being extended by another class. That's why it's common for it to be included in such cases.

I added comments to show how everything works. Here is the finalized result:

```haxe
package funkin.menus;

// Import the VideoCutscene class for use.
import funkin.game.cutscenes.VideoCutscene;

// Begin the class, extending VideoCutscene.
class PeterCutsceneIntro extends VideoCutscene {
    public function new() {
	// Call super() just as you would in a class with an extended class.
	// Make sure that the video file in the first argument is named correctly, otherwise it likely will not work.
	// For the second argument, switch states as you normally would.
        super(Paths.video("betergriffin", "mp4"), FlxG.switchState(new TitleState()))
    };
}

// EOF
``` 

If there seem to be any more problems, please tell me. I haven't tested this, so there may be some small errors.

**Reply by THEkatinamicrowave** [reply]
There were a couple errors
```
source/funkin/menus/PeterCutsceneIntro.hx:12: characters 51-85 : Void should be () -> Void
source/funkin/menus/PeterCutsceneIntro.hx:12: characters 51-85 : ... For function argument 'callback'
source/funkin/menus/PeterCutsceneIntro.hx:8: lines 8-13 : Missing super constructor call
```
I managed to fix these by separating the FlxG switchstate into a separate function as shown here
![image](https://github.com/FNF-CNE-Devs/CodenameEngine/assets/89645987/5398ab32-dd1f-44d3-b36d-f494f212d560)
unfortunately, it still returns a Null Object Error upon loading up PeterCutsceneIntro.hx.
I looked in VideoCutscene.hx and found this so I'm wondering if maybe I could find a way to use that startVideo function
I'll do it after posting this to see if it works but yeah until then here's what I found.
![image](https://github.com/FNF-CNE-Devs/CodenameEngine/assets/89645987/32851ebd-b6c0-4b00-bdf7-e04b4f9ff512)

**Reply by THEkatinamicrowave** [reply]
Also yeah the file name is "betergriffin.mp4" so that can't be the issue (I checked)

**Reply by THEkatinamicrowave** [reply]
While we fix this, I'll make another build called -debugplusgame so that I can actually... yknow... play the game.

**Reply by r6915ee** [reply, marked as answer]
Ah, it might be on my end, then. I don't usually interact with substates, but since `BetaWarningState` is supposed to open up a state and not a substate, it might not work by default. I'd recommend importing `flixel.FlxSubState` in that case into `BetaWarningState`, and then replace your `FlxG.switchState(new PeterCutsceneIntro())` with `openSubState(new PeterCutsceneIntro())` instead.

**Reply by THEkatinamicrowave** [reply]
THANK YOU SO MUCH THAT TOTALLY WORKED !!!!!!!!!!!!!
![image](https://github.com/FNF-CNE-Devs/CodenameEngine/assets/89645987/665c0076-0459-40ce-a6e8-548c65a66d01)
I'll edit TitleState for the camera error thing but TYSM!!!!!!!!! U r best bro I will put you in credits (please put your credits icon in the replies)

**Reply by r6915ee** [reply]
No problem! Crediting isn't necessary, but if you want to then here is my icon. GitHub integration could work too.
![scheese](https://github.com/FNF-CNE-Devs/CodenameEngine/assets/111615291/8f1d2de2-3dc2-4f55-b787-0ddcb1e4264f)
